src/main/java/com/jwplayer/jwplatform/client/AdvertisingClient.java

```diff
@@ -0,0 +1,143 @@
+package com.jwplayer.jwplatform.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.json.JSONObject;
+
+import com.google.common.base.Preconditions;
+import com.jwplayer.jwplatform.exception.JWPlatformException;
+import com.jwplayer.jwplatform.rest.HttpCalls;
+
+/**
+ * JW Platform Advertising API client.
+ *
+ * <p>An API client for the JW Platform Advertising API. For the API documentation see:
+ * https://developer.jwplayer.com/jwplayer/reference#introduction-to-api-v2
+ *
+ * <p>Example:
+ *    AdvertisingClient client = AdvertisingClient.getClient(secret);
+ */
+public class AdvertisingClient{
```

Comment:
You can create an abstract class that each of these clients extends that contains addHeader/removeHeader so that you don't need to repeat those functions in every one of these Clients.